### PR TITLE
feat: add kinesis stream support

### PR DESCRIPTION
**Pull request by strongishllama**
## Description

This pull request adds support for nuking Kinesis Stream as mentioned in #301. I'll bring in Kinesis Firehose support in another pull request when I get a bit more time.

I pretty much just translated the work done in this [pull request](https://github.com/gruntwork-io/cloud-nuke/pull/289/files) and applied it to Kinesis Streams so please let me know if anything is incorrect style/logic wise 😄 

Tests in the `kinesis_stream_test.go` are passing as well as when you run the CLI with `go run main.go aws --resource-type kinesis-stream`.

Docs have been updated to show the Kinesis Stream support.

Because this is adding support for a new resource this pull request will introduce breaking changes as per [here](https://github.com/gruntwork-io/cloud-nuke/blob/a696846eadb65719badc5bc56706a9c8a9b91292/README.md#choosing-a-new-release-tag)

### CODE REVIEW

**Comment by strongishllama** [comment]
My apologies, I've just had a closer look at the diff and it seems that `gofumpt` has made a couple of unintended style changes. I'll revert them now!

File: aws/kinesis_stream_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+func deleteKinesisStream(t *testing.T, svc *kinesis.Kinesis, name *string, checkErr bool) {
+	_, err := svc.DeleteStream(&kinesis.DeleteStreamInput{
+		StreamName: name,
+	})
+	if checkErr {
+		require.NoError(t, err)
+	}
+}
+
+func assertKinesisStreamsDeleted(t *testing.T, svc *kinesis.Kinesis, identifiers []*string) {
```
[the remark this answers is not in the document]
Comment (strongishllama) [reply aws/kinesis_stream_test.go:110]: Ah, good catch! Added the missing calls 👍

**Comment by strongishllama** [comment]
@denis256 👋 Just checking in to see if you've had a chance to look at this yet? 😄

**Comment by denis256** [comment]
Hi,
will trigger a CI build to see if all tests pass

**Comment by denis256** [comment]
```
=== CONT  TestNukeKinesisStreamOne
    kinesis_stream_test.go:116: 
        	Error Trace:	kinesis_stream_test.go:116
        	            				kinesis_stream_test.go:57
        	Error:      	Expected value not to be nil.
        	Test:       	TestNukeKinesisStreamOne
markers (0 deletion markers) from bucket cloud-nuke-test-p9bgvlvn55pl"
--- FAIL: TestNukeKinesisStreamOne (15.79s)

=== CONT  TestNukeKinesisStreamMoreThanOne
    kinesis_stream_test.go:116: 
        	Error Trace:	kinesis_stream_test.go:116
        	            				kinesis_stream_test.go:83
        	Error:      	Expected value not to be nil.
        	Test:       	TestNukeKinesisStreamMoreThanOne
--- FAIL: TestNukeKinesisStreamMoreThanOne (48.14s)

```

**Comment by strongishllama** [comment]
@denis256 My bad, small logic error. Should be good to go now.

**Comment by denis256** [comment]
Hi,
after re-run tests seems to pass

**Comment by strongishllama** [comment]
@denis256 Conflicts have been fixed 👍

**Comment by strongishllama** [comment]
@denis256 Sorry to keep nagging but do you think this is good to go now? 😅

**Comment by denis256** [comment]
Looks like after merge build is failing:
```
Stderr: go: downloading github.com/konsorten/go-windows-terminal-sequences v1.0.3
# github.com/gruntwork-io/cloud-nuke/aws
aws/aws.go:724:40: use of package session without selector
```

**Comment by strongishllama** [comment]
> Looks like after merge build is failing:
> 
> ```
> Stderr: go: downloading github.com/konsorten/go-windows-terminal-sequences v1.0.3
> # github.com/gruntwork-io/cloud-nuke/aws
> aws/aws.go:724:40: use of package session without selector
> ```

Fixed 👍

**Comment by denis256** [comment]
Hi,
looks like with recent changes are conflicting files:
```
README.md
aws/aws.go
config/config.go
config/config_test.go
```

**Comment by strongishllama** [comment]
> Hi, looks like with recent changes are conflicting files:
> 
> ```
> README.md
> aws/aws.go
> config/config.go
> config/config_test.go
> ```

Resolved 👍

**Review by denis256 — APPROVED** [review]
CI job passed without failures, going to merge into master

**Pull request merged**